src/MiniProfiler.Shared/Timing.cs

```diff
@@ -253,19 +258,44 @@ public void Stop()
         /// </remarks>
         public void AddChild(Timing timing)
         {
-            if (Children == null)
+            if (Interlocked.Increment(ref _threadsAccessing) > 1)
+                throw new Exception("Concurrent access");
+
+            Task.Delay(10).Wait();
```

Comment:
Thread.Sleep I guess is not available in .netcore (or perhaps i need to install a separate package).